widgetssdk/src/main/java/com/glia/widgets/launcher/EngagementLauncher.kt

```diff
@@ -46,10 +49,18 @@ interface EngagementLauncher {
 internal class EngagementLauncherImpl(
     private val activityLauncher: ActivityLauncher,
     private val hasOngoingSecureConversationUseCase: HasOngoingSecureConversationUseCase,
-    private val configurationManager: ConfigurationManager
+    private val isQueueingOrLiveEngagementUseCase: IsQueueingOrLiveEngagementUseCase,
+    private val endEngagementUseCase: EndEngagementUseCase,
+    private val configurationManager: ConfigurationManager,
+    private val controllerFactory: ControllerFactory
 ) : EngagementLauncher {
 
     override fun startChat(context: Context, visitorContextAssetId: String?) {
+        if (isQueueingOrLiveEngagementUseCase.isQueueing) {
+            endEngagementUseCase()
+            controllerFactory.destroyChatController()
+        }
```

Comment:
Can we move this to a separate function to avoid code duplication?

Reply:
@DavDo I also thought about this. However, only two of them are identical: some call `destroyChatController()` and some `destroyCallController()`.

Reply:
Sorry, I haven't noticed that difference.